packages/blockchain-wallet-v4-frontend/src/layouts/Wallet/MenuTop/Balance/index.js

```diff
@@ -1,25 +1,17 @@
 import React from 'react'
 import { connect } from 'react-redux'
-import { bindActionCreators } from 'redux'
 
-import { actions } from 'data'
-import { getData } from './selectors'
+import { selectors } from 'data'
 import Template from './template'
 
 class Balance extends React.PureComponent {
   render () {
-    const { data } = this.props
-    return <Template btcUnspendableContext={data.btcUnspendableContext} bchUnspendableContext={data.bchUnspendableContext} path={data.path} />
+    return <Template path={this.props.path} />
```

Comment:
Something is wrong here:
![screen shot 2018-06-20 at 10 13 12 am](https://user-images.githubusercontent.com/5640772/41663930-917e9a8e-7472-11e8-97b6-7e325841bec8.png)

Reply:
As explained earlier, there was a bug in the way we calculate the getSpendableContext and getUnspendableContext for BCH, and it was automatically excluding the legacyAddresses from the wallet payload. To double check carefully then, but these legacyAddresses were previously not included in the Spendable/Unspendable context (if not archived) and were influencing the final balance.

Reply:
Ok, I misunderstood the concept of Unspent balances and thought it was only about 'Imported addresses'. check my last changes